Refactor spec file handling

mikemckiernan opened the pull request:
For a start.  I can follow up with more detail.

* Appending to `html_static_path` didn't do what I thought it did.  It coincidentally worked for tests.
* Preserve the source directory structure in the build output so that we sidestep the risk of clobbering files with the same name.
* Calculate the path from the HTML file to the `_static` directory more carefully.  Add a test that uses two subdirs at different depths.

kasium commented:
Can you please add an integration test?

mikemckiernan commented:
> Can you please add an integration test?

PLMK if what I added meets your expectations and any other comments or corrections. Thanks very, very much!

kasium commented:
Notes for myself

* the build dir should be a tmp dir
* Copy the spec on demand, reduces the code size

kasium commented:
Just one question before we can merge it: have you considered to just use the existing test doc folder structure?

No strong opinion here just wondering

mikemckiernan commented:
> Just one question before we can merge it: have you considered to just use the existing test doc folder structure?
> 
> No strong opinion here just wondering

I’m in no rush.  I added a second “test root” to make clear that existing tests ran without modification—I thought that was more valuable than optimization. However, I have no objection to modifying or combining the tests in any way. If anything, I was glad for your integration test request because the unit test that I added requires a lot of imagination to keep track of the files added and the specifications that are referenced.  

Again, I’m in no rush and I’m happy to revise toward the clearest, most maintainable code and tests possible.

kasium commented:
Thanks. Lets get this merged
